unitTests/ui/FieldValueTest.ts

```diff
@@ -32,25 +31,43 @@ export function FieldValueTest() {
       initializeFieldValueComponent({ field: '@string' });
     });
 
-    function initializeFieldValueComponent(
+    function initializeFieldValueComponent<T extends Facet | DynamicFacet>(
       options: IFieldValueOptions,
       result = FakeResults.createFakeResult(),
-      facet?: Facet | DynamicFacet
+      ...facets: T[]
     ) {
       result.raw.filetype = 'unknown';
       test = Mock.advancedResultComponentSetup<FieldValue>(FieldValue, result, <Mock.AdvancedComponentSetupOptions>{
         element: $$('span').el,
         cmpOptions: options,
         modifyBuilder: (builder: Mock.MockEnvironmentBuilder) => {
-          if (facet) {
-            builder.componentStateModel.get = () => [facet];
+          if (facets.length) {
+            builder.componentStateModel.get = () => facets;
             builder.queryStateModel.get = () => [];
+            facets.forEach(facet => builder.searchInterface.attachComponent((<typeof Facet>facet.constructor).ID, facet));
           }
           return builder;
         }
       });
     }
 
+    function initializeFacet<T extends Facet | DynamicFacet>(type: { new (...args: any[]): T; ID: string }, options: T['options']) {
+      const newFacet = Mock.mockComponent<T>(type);
+
+      newFacet.constructor['ID'] = 'Facet';
```

Comment:
Could this be DynamicFacet sometimes as well?

Reply:
Woops, my bad, it was meant to take the correct ID.

Thanks for finding it and pointing it out!